assets/js/blocks/reviews-by-product/editor-block.js

```diff
@@ -0,0 +1,104 @@
+/**
+ * External dependencies
+ */
+import { Component, Fragment } from 'react';
+import PropTypes from 'prop-types';
+import { debounce } from 'lodash';
+
+/**
+ * Internal dependencies
+ */
+import { getOrderArgs, getReviews } from './utils';
+import LoadMoreButton from '../../frontend/components/load-more-button';
+import OrderBySelect from '../../frontend/components/order-by-select';
+import Reviews from '../../frontend/components/reviews';
+import withComponentId from '../../frontend/hocs/with-component-id';
+
+/**
+ * Block rendered in the editor.
+ */
+class EditorBlock extends Component {
+	constructor() {
+		super( ...arguments );
+
+		this.state = {
+			reviews: [],
+			totalReviews: 0,
+		};
+
+		this.debouncedLoadFirstReviews = debounce( this.loadFirstReviews.bind( this ), 400 );
+	}
+
+	componentDidMount() {
+		this.loadFirstReviews();
+	}
+
+	componentDidUpdate( prevProps ) {
+		if (
+			prevProps.attributes.orderby !== this.props.attributes.orderby ||
+			prevProps.attributes.productId !== this.props.attributes.productId ||
+      prevProps.attributes.reviewsOnPageLoad !== this.props.attributes.reviewsOnPageLoad
+		) {
+			this.debouncedLoadFirstReviews();
+		}
+	}
+
+	getDefaultArgs() {
+		const { attributes } = this.props;
+		const { order, orderby } = getOrderArgs( attributes.orderby );
+		const { productId, reviewsOnPageLoad } = attributes;
+
+		return {
+			order,
+			orderby,
+			per_page: reviewsOnPageLoad,
+			product_id: productId,
+		};
+	}
+
+	loadFirstReviews() {
+		const args = this.getDefaultArgs();
+
+		getReviews( args ).then( ( { reviews, totalReviews } ) => {
+			this.setState( { reviews, totalReviews } );
+		} ).catch( () => {
+			this.setState( { reviews: [] } );
+		} );
+	}
+
+	render() {
+		const { attributes, componentId } = this.props;
+		const { reviews, totalReviews } = this.state;
+
+		return (
+			<Fragment>
+				{ ( attributes.showOrderby && wc_product_block_data.enableReviewRating ) && (
```

Comment:
where does `wc_product_block_data` come from?  Is this a global variable?  If so, it'd be preferable to have all global variables defined in a module that gets imported where it's used, so defaults when not present can be handled.  

Short term, if you want to avoid doing that here, you could set the variable at the top of the file so it can be safely used here if for whatever reason it's not available.  So something like this right after the imports:

```js
const enableReviewRating = !! ( typeof wc_product_block_data !== 'undefined' && wc_product-block_data.enableReviewRating );
```

Reply:
Can we do this in a follow up PR so we don't hold up development of the other review blocks?

Reply:
I like the idea of moving it into a module, but since `wc_product_block_data` is currently used in many files other than the ones of this PR, I would prefer doing that in another PR.

For now, I moved the access to `wc_product_block_data` to the top of the file as you suggested: 0ba5079aa22af6f5f86f8219fcf020332db783c6.